shared/types/field/base.ts

```diff
@@ -10,6 +10,7 @@ export enum BasicField {
   ShortText = 'textfield',
   LongText = 'textarea',
   Dropdown = 'dropdown',
+  Country = 'country',
```

Comment:
As discussed, this PR should not have any knowledge of `BasicField.Country`; just create a `CountryField` component that is basically a `BasicField.Dropdown` with hardcoded country field options

Reply:
Addressed in https://github.com/opengovsg/FormSG/pull/3705